pkg/filesystem/fs/fs.go

```diff
@@ -844,6 +864,9 @@ func (fs *Filesystem) createSharedDaemon(snapshotID string, imageID string) (*da
 		return nil, err
 	}
 
+	// FIXME: It's a little tricky
+	d.Supervisor = fs.manager.SupervisorSet.GetSupervisor("shared_daemon")
```

Comment:
Why not put this into `WithSupervisor` here?

Reply:
Because a supervisor needs a unique ID when creating. In order to make it easy to trace and debug, I am using the daemon ID as the supervisor ID. We can only get the daemon ID after daemon is created.